src/include/planner/planner.h

```diff
@@ -26,6 +27,37 @@ namespace planner {
 
 struct LogicalInsertInfo;
 
+enum class SubqueryType : uint8_t {
+    NONE = 0,
+    INTERNAL_ID_CORRELATED = 1,
+    CORRELATED = 2,
+};
+
+struct QueryGraphPlanningInfo {
+    // Predicate info.
+    binder::expression_vector predicates;
+    // Subquery info.
+    SubqueryType subqueryType = SubqueryType::NONE;
+    binder::expression_vector corrExprs;
+    cardinality_t corrExprsCard = 0;
+    // Join hint info.
+    std::shared_ptr<binder::BoundJoinHintNode> hint = nullptr;
```

Comment:
Can we use unique pointer there?

Reply:
Yes for this PR but not in the soon future where we will need shared_ptr in DP table.